### PR TITLE
Standardizing Duckdb connector documentation

### LINKED ISSUE #587: Enhance DuckDB connector documentation
Page should have:
- [ ] Connector description
- [ ] Configuration section
- [ ] Secrets section / Authentication section
- [ ] Examples section
- [ ] Parameters table
- [ ] File format support (if applicable)

### PR DESCRIPTION
**Pull request by eadgbear**
## 🗣 Description

Standardizing and enhancing DuckDB connector documentation as outlined in #580 
## 🔨 Related Issues

- Closes #587

### CODE REVIEW

**Review by sgrebnov — DISMISSED** [review]

File: spiceaidocs/docs/components/data-connectors/duckdb.md [hunk 1]
```diff
@@ -4,9 +4,9 @@ sidebar_label: 'DuckDB Data Connector'
 description: 'DuckDB Data Connector Documentation'
 ---
 
-## Dataset Source
+DuckDB is an in-process SQL OLAP database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.
 
-To connect to a DuckDB [persistent database](https://duckdb.org/docs/connect/overview#persistent-database) as a data source, specify `duckdb` as the selector in the `from` value for the dataset.
+This connector allows DuckDB [persistent database](https://duckdb.org/docs/connect/overview#persistent-database) to be used as a data source for federated SQL queries.
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/duckdb.md]:
```suggestion
This connector allows DuckDB [persistent databases](https://duckdb.org/docs/connect/overview#persistent-database) to be used as a data source for federated/accelerated SQL queries.
```

File: spiceaidocs/docs/components/data-connectors/duckdb.md [hunk 2]
```diff
@@ -81,3 +121,7 @@ SELECT * FROM 'todos.json';
 -- As a DuckDB function
 SELECT * FROM read_json('todos.json');
 ```
+
+## Secrets
```
Comment (phillipleblanc) [inline spiceaidocs/docs/components/data-connectors/duckdb.md]: Does the DuckDB data connector need any secrets?

File: spiceaidocs/docs/components/data-connectors/duckdb.md [hunk 3]
```diff
@@ -18,13 +18,53 @@ datasets:
...
 
+### `from`
+
+The `from` field supports one of two forms:
+
+| `from`                         | Description                                                                                                                                             |
+| ------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `duckdb:database.schema.table` | Read data from a table named `database.schema.table` in the DuckDB file                                                                                |
+| `duckdb:read_*()`              | Read data using one of the common [data import](https://duckdb.org/docs/data/overview) DuckDB functions, e.g `read_json`, `read_parquet` or `read_csv`. |
+
```
Comment (Sevenannn) [inline spiceaidocs/docs/components/data-connectors/duckdb.md:29]: DuckDB functions are not restricted to read_*(), functions like glob('*') can also be passed in, e.g.
```YAML
  - from: duckdb:glob('*')
    name: hits
```

Comment (phillipleblanc) [reply spiceaidocs/docs/components/data-connectors/duckdb.md:29]: Yes, any DuckDB functions that create a table/result set can be used

Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/duckdb.md:29]: Corrected in the latest push!

Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/duckdb.md:29]: Also I learned something new about the product

File: spiceaidocs/docs/components/data-connectors/duckdb.md [hunk 4]
```diff
@@ -4,9 +4,9 @@ sidebar_label: 'DuckDB Data Connector'
 description: 'DuckDB Data Connector Documentation'
 ---
 
-## Dataset Source
+DuckDB is an in-process SQL OLAP database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.
 
-To connect to a DuckDB [persistent database](https://duckdb.org/docs/connect/overview#persistent-database) as a data source, specify `duckdb` as the selector in the `from` value for the dataset.
+This connector allows DuckDB [persistent databases](https://duckdb.org/docs/connect/overview#persistent-database) to be used as a data source for federated/accelerated SQL queries.
```
Comment (peasee) [inline spiceaidocs/docs/components/data-connectors/duckdb.md]:
```suggestion
This connector supports DuckDB [persistent databases](https://duckdb.org/docs/connect/overview#persistent-database) as a data source for federated SQL queries.
```

File: spiceaidocs/docs/components/data-connectors/duckdb.md [hunk 5]
```diff
@@ -4,9 +4,9 @@ sidebar_label: 'DuckDB Data Connector'
 description: 'DuckDB Data Connector Documentation'
 ---
 
-## Dataset Source
+DuckDB is an in-process SQL OLAP database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.
```
Comment (peasee) [inline spiceaidocs/docs/components/data-connectors/duckdb.md]: What is OLAP?

File: spiceaidocs/docs/components/data-connectors/duckdb.md [hunk 6]
```diff
@@ -18,13 +18,53 @@ datasets:
...
+```
+
+### `params`
+
 The DuckDB data connector can be configured by providing the following `params`:
 
-- `duckdb_open`: The name for the file to back the DuckDB database.
+| Parameter Name | Description                                        |
+| -------------- | -------------------------------------------------- |
+| `duckdb_open`  | The name for the file to back the DuckDB database. |
```
Comment (peasee) [inline spiceaidocs/docs/components/data-connectors/duckdb.md]:
```suggestion
| `duckdb_open`  | The name of the DuckDB database to open. |
```

File: spiceaidocs/docs/components/data-connectors/duckdb.md (hunk 5 above)
Comment (Sevenannn) [inline spiceaidocs/docs/components/data-connectors/duckdb.md]:
```suggestion
DuckDB is an in-process SQL OLAP (Online Analytical Processing) database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.
```

Comment (Sevenannn) [reply spiceaidocs/docs/components/data-connectors/duckdb.md]: Added in the newest version: Online Analytical Processing

**Review by Sevenannn — DISMISSED** [review]

**Review by lukekim — APPROVED** [review]

**Pull request merged**